superset-frontend/src/explore/components/DatasourcePanel/index.tsx

```diff
@@ -203,45 +258,57 @@ export default function DataSourcePanel({
             header={<span className="header">{t('Metrics')}</span>}
             key="metrics"
           >
-            <div className="field-length">
-              {t(`Showing %s`, lists.metrics.length)}
-            </div>
-            {lists.metrics.map(m => (
-              <LabelContainer key={m.metric_name} className="column">
-                {enableExploreDnd ? (
-                  <DatasourcePanelDragWrapper
-                    value={m}
-                    type={DndItemType.Metric}
-                  >
-                    <MetricOption metric={m} showType />
-                  </DatasourcePanelDragWrapper>
-                ) : (
-                  <MetricOption metric={m} showType />
-                )}
-              </LabelContainer>
-            ))}
+          { showAllMetrics ?
+            <>
+              <div className="field-length">
+                {t(`Showing %s of %s`, lists.metrics.length, lists.metrics.length)}
+              </div>
+              <DisplayMetrics metricList={lists.metrics} enableExploreDnd={enableExploreDnd} />
+              <button onClick={() => setShowAllMetrics(false)}>
+                Show less...
+              </button>
+            </>
+            :
+            <>
+              <div className="field-length">
+                {t(`Showing %s of %s`, metricSlice.length, lists.metrics.length)}
+              </div>
+              <DisplayMetrics metricList={metricSlice} enableExploreDnd={enableExploreDnd} />
+              { lists.metrics.length > 50 ?
+                <button onClick={() => setShowAllMetrics(true)}>
+                  Show more...
```

Comment:
I agree that "Show more" is kind of the opposite of "Show less" but it makes me thing we might be showing another 50 columns rather than all of them. We might want to keep this as "Show all"?

Reply:
> 
> 
> I agree that "Show more" is kind of the opposite of "Show less" but it makes me thing we might be showing another 50 columns rather than all of them. We might want to keep this as "Show all"?

Yeah, that's a good point. I'll change that.